math/src/field/f64/mod.rs

```diff
@@ -319,14 +355,8 @@ impl DivAssign for BaseElement {
 impl Neg for BaseElement {
     type Output = Self;
 
-    #[inline]
     fn neg(self) -> Self {
-        let v = self.as_int();
-        if v == 0 {
-            Self::ZERO
-        } else {
-            Self(M - v)
-        }
+        Self::sub(BaseElement::ZERO, self)
```

Comment:
We should probably change the body to:
```Rust
Self::ZERO - self
```
Also, I'd keep the `#[inline]` attribute.